### PR TITLE
Support Nexus operation complete before start

### PR DESCRIPTION
**Pull request by pdoerner**
## What changed?

Added support for completing Nexus operations before the StartOperation response is received. Operation ID, StartTime, and StartLinks were added to the completion request so that a NexusOperationStarted event can be fabricated.
Depends on https://github.com/nexus-rpc/sdk-go/pull/26 and https://github.com/temporalio/sdk-go/pull/1710

Bumps Temporal SDK version to `v1.30.1`
Bumps Nexus SDK version to `v0.0.12`

## Why?

Better user experience.

## How did you test it?

New functional test

### CODE REVIEW

File: components/nexusoperations/completion.go
Comment (bergundy) [inline components/nexusoperations/completion.go:141]: nit:

```suggestion
	operation, err := hsm.MachineData[Operation](node)
```

File: components/nexusoperations/completion.go
Comment (bergundy) [inline components/nexusoperations/completion.go:148]: Not sure I understand why you're returning not found here.

File: components/nexusoperations/completion.go
Comment (bergundy) [inline components/nexusoperations/completion.go:166]: I'm not sure this is something that's supported or what the implications are. @yycptt, @alexshtin?

File: components/nexusoperations/frontend/handler.go
Comment (bergundy) [inline components/nexusoperations/frontend/handler.go:189]: Wondering if we want to warn instead.

File: components/nexusoperations/frontend/handler.go
Comment (bergundy) [inline components/nexusoperations/frontend/handler.go:202]: Same here.

File: service/history/workflow/mutable_state_impl.go
Comment (bergundy) [inline service/history/workflow/mutable_state_impl.go:583]: This should be the namespace name, not ID. We don't have APIs that can interact with namespace IDs.

File: tests/nexus_workflow_test.go
Comment (bergundy) [inline tests/nexus_workflow_test.go:788]: Would have been ideal to test this with the Go SDK and the WorkflowRunOperation. You can wrap the handler to delay sending the response until the caller workflow completes.

File: tests/nexus_workflow_test.go
Comment (bergundy) [inline tests/nexus_workflow_test.go:839]: AFAIR using suite methods from background routines may panic. Don't remember the details though.

**Review by bergundy — COMMENTED** [review]
Mostly LGTM. Added some small comments.

File: components/nexusoperations/completion.go
Comment (pdoerner) [reply components/nexusoperations/completion.go:166]: I will leave this out of this PR for now and add `StartTime` as an attribute in a followup PR.

File: components/nexusoperations/completion.go
Comment (pdoerner) [reply components/nexusoperations/completion.go:148]: I did that because that's what we are returning in other places where a transition is not possible, but thinking about it more I'm not sure we even need to error here.

File: components/nexusoperations/completion.go
Comment (bergundy) [inline components/nexusoperations/completion.go:164]: Better use the transition function.

File: tests/nexus_workflow_test.go
Comment (bergundy) [inline tests/nexus_workflow_test.go:737]: Why do you need the cancel function here?

Comment (bergundy) [reply tests/nexus_workflow_test.go:737]: This test would be much more readable if you used the SDK. Please change.

File: tests/nexus_workflow_test.go
Comment (bergundy) [inline tests/nexus_workflow_test.go:822]: You don't need this. Just fail the request.

**Review by bergundy — APPROVED** [review]
Overall LGTM.

The test could be improved. I'd like to see that done before merging but not blocking the PR.

File: tests/nexus_workflow_test.go
Comment (pdoerner) [reply tests/nexus_workflow_test.go:822]: I wanted to test that the server handles getting the started response appropriately when they are received out of order. Failing the request means we would never see the started response which doesn't quite match the real-world race condition.

File: tests/nexus_workflow_test.go
Comment (pdoerner) [reply tests/nexus_workflow_test.go:737]: I added the cancel function because I wanted to make sure that the context passed to the background routine is canceled if the test exits early because of an assertion failure.

Comment (pdoerner) [reply tests/nexus_workflow_test.go:737]: I will rewrite the test to use the SDK in a followup.

**Pull request merged**
